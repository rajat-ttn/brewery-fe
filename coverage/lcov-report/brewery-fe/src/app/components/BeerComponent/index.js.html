<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for brewery-fe/src/app/components/BeerComponent/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">brewery-fe/src/app/components/BeerComponent</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { PropTypes } from 'prop-types';
import { convertTempCelciusToFahrenheit } from '../../../util/helper';
import './style.css';
&nbsp;
const BeerComponent = ({ beerContentDetail, temperatureType }) =&gt; {
    const { beerType, tempRange, currentTemperature } = beerContentDetail;
    const minTemp = convertTempCelciusToFahrenheit(temperatureType,tempRange[0]);
    const maxTemp = convertTempCelciusToFahrenheit(temperatureType,tempRange[1]);
    const parsedCurrTemp = parseInt(currentTemperature,10);
    const tempSymbol = temperatureType === 'CELSIUS' ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;span&gt;&amp;#8451;&lt;/span&gt; </span>: &lt;span&gt;&amp;#8457;&lt;/span&gt;;
    const alertCss = parsedCurrTemp &gt; maxTemp ? <span class="branch-0 cbranch-no" title="branch not covered" >'danger red ' </span>: currentTemperature &lt; minTemp ? <span class="branch-0 cbranch-no" title="branch not covered" >'danger blue' </span>: '';
    const showCurrentTemp = parsedCurrTemp ? <span class="branch-0 cbranch-no" title="branch not covered" >convertTempCelciusToFahrenheit(temperatureType,parsedCurrTemp) </span>: '--';
    const showSymbol = parsedCurrTemp ? <span class="branch-0 cbranch-no" title="branch not covered" >tempSymbol </span>: '';
    return (
         &lt;div className="col-md-4 commonBoxModel"&gt;
              &lt;div className="beerContainer"&gt;
                &lt;div className="overlay"&gt;
                    &lt;h3 className="beertype"&gt;{beerType}&lt;/h3&gt;
                        &lt;span className={`currentTemp ${alertCss}`}&gt;
                                &lt;img className="iconStyling" src="../images/temp-icon.png" /&gt;
                            {showCurrentTemp } {showSymbol}
                        &lt;/span&gt;
                       &lt;span className="tmpLabel"&gt;Current Temp&lt;/span&gt;
&nbsp;
                       &lt;span className="rangeTemp"&gt;
                           &lt;img className="iconStyling" src="../images/temp-icon.png" /&gt;
                           {`${minTemp} - ${maxTemp}`}{tempSymbol}
                       &lt;/span&gt;
&nbsp;
                       &lt;span className="tmpLabel"&gt;Temp Range&lt;/span&gt;
                &lt;/div&gt;
              &lt;/div&gt;
          &lt;/div&gt;
    );
};
BeerComponent.propTypes = {
    beerContentDetail: PropTypes.object.isRequired,
    temperatureType: PropTypes.string.isRequired,
};
&nbsp;
export default BeerComponent;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 10 2018 17:04:54 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
