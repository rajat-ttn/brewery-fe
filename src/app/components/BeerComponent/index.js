import React from 'react';
import { PropTypes } from 'prop-types';
import { ConvertTempCelciusToFahrenheit } from '../../../util/helper';
import './style.css';

const BeerComponent = ({ beerContentDetail, temperatureType }) => {
    const { beerType, tempRange, currentTemperature } = beerContentDetail;
    const minTemp = ConvertTempCelciusToFahrenheit(temperatureType,tempRange[0]);
    const maxTemp = ConvertTempCelciusToFahrenheit(temperatureType,tempRange[1]);
    const tempSymbol = temperatureType === 'CELSIUS' ? `&#8451;` : `&#8457;`;
    return (
         <div className="col-md-4 commonBoxModel">
              <div className="beerContainer">
                   <h3 className="beertype">{beerType}</h3>
                   <span className="currentTemp danger">
                       <img className="iconStyling" src="../images/temp-icon.png" />
                       {currentTemperature ? ConvertTempCelciusToFahrenheit(temperatureType,currentTemperature) : '--'}&#8451;
                   </span>
                   <span className="tmpLabel">Current Temp</span>
                   <span className="rangeTemp">
                       <img className="iconStyling" src="../images/temp-icon.png" />
                       {`${minTemp} - ${maxTemp}`} &#8451;
                       </span>
                   <span className="tmpLabel">Temp Range</span>
              </div>
          </div>
    );
};
BeerComponent.propTypes = {
    beerContentDetail: PropTypes.object.isRequired,
    temperatureType: PropTypes.string.isRequired,
};

export default BeerComponent;