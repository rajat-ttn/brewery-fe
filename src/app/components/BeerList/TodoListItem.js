import React from 'react';
import { connect } from 'react-redux';
import * as actions from './actionCreators';

function TodoListItem(props){
    return (
        <li>
            <small className={'is-completed'}>
                <input type='checkbox'
                       checked={props.item.isCompleted}
                       onChange={ (event) => {
                           props.updateTodo({
                               ...props.item,
                               isCompleted: event.target.checked
                           });
                       }  }>
                </input>
            </small>
            <span className={ (props.item.isCompleted ?'striked': null)  +  ' todo-text'}>{props.item.text}</span>
            <small
                onClick={() => { props.deleteTodo(props.item) }}
                className={'delete'}>
                X
            </small>
        </li>
    )
}


export default connect(null, actions)(TodoListItem);