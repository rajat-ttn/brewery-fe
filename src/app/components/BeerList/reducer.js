// import ACTION from '../../../constants/actions';
var beerData =  [
    {
        beerType: 'lorem',
        beerImageUrl: 'data.png',
        range: [4, 6],
        currentTemp: 5
    },
    {
        beerType: 'lorem2',
        beerImageUrl: 'data.png',
        range: [4, 6],
        currentTemp: 5
    },
    {
        beerType: 'lorem3',
        beerImageUrl: 'data.png',
        range: [4, 6],
        currentTemp: 5
    },
    {
        beerType: 'lorem4',
        beerImageUrl: 'data.png',
        range: [4, 6],
        currentTemp: 5
    },
];

export default function(state = beerData, action) {
    return  state;
};
