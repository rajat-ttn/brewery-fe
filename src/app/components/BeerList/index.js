import React from 'react';
import { PropTypes } from 'prop-types'
import BeerComponent from '../BeerComponent'

const BeerList = ({ beerList, temperatureType }) => {
    return (
        <div className="row beerlistSection">
            {
                beerList && beerList.length ?
                    beerList.map((beer, index) => (
                        <BeerComponent beerContentDetail={beer} key={beer.id} temperatureType={temperatureType} />
                    )) : <div className="msgStyle"><p>No Container Found</p></div>
            }
        </div>
    );
};

BeerList.propTypes = {
    beerList: PropTypes.array.isRequired,
    temperatureType: PropTypes.string.isRequired,
};
export default BeerList;