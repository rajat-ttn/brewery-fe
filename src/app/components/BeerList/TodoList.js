import React, { Component } from 'react';
import { connect } from 'react-redux';
import TodoListItem from './TodoListItem';

class TodoList extends Component {
    render() {
        return (
            <div className="todo-list">
                <ul>
                    {this._renderListItems()}
                </ul>
            </div>
        );
    }
    _renderListItems = () => {
        return this.props.todoItems.map((item, index)=>{
            return <TodoListItem item={item} key={item.id} />
        })
    }
}

function mapStateToProps(state, ownProps){
    let filteredList;
    switch(state.filter){
        case 'ALL': filteredList = state.todoList;
            break;
        case 'COMPLETED': filteredList = state.todoList.filter((item) => item.isCompleted  );
            break;
        case 'ACTIVE': filteredList = state.todoList.filter((item) => !item.isCompleted  );
            break;
    }
    return {
        todoItems : filteredList
    }
}

export default connect(mapStateToProps)(TodoList);