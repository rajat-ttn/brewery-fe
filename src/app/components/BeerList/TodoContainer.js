import React, { Component } from 'react';

import AddTodo from './AddTodo';
import TodoList from './TodoList';
import Filter from './Filter';

class TodoContainer extends Component {
    render() {
        return (
            <div className="todo-container">
                <AddTodo></AddTodo>
                <TodoList></TodoList>
                <Filter></Filter>
            </div>
        );
    }
}

export default TodoContainer;