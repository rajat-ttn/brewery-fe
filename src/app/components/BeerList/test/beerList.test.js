import React from 'react';
import { shallow } from 'enzyme';
import BeerList from '../index';
import BeerComponent from '../../BeerComponent/index';


describe('BeerList Component should render correctly', () => {
    let allProps,
        beerListContainer;

    beforeEach(() => {
        allProps = {
            beerList: [{beerType : "Aler Beer", containerId: 1, id: 1, tempRange: [3,5]}],
            temperatureType: "FAHRENHEIT",
        };
        beerListContainer = shallow(<BeerList {...allProps}  />);
    });

    describe("render `BeerComponent` when beerlist is defined", () => {
        it("always renders a `BeerComponent`", () => {
            expect(beerListContainer.find(BeerComponent).length).toBe(1);
        });

        it("BeerComponent must need to pass 2 props ", () => {
            const beerComponent = beerListContainer.find(BeerComponent);
            expect(Object.keys(beerComponent.props()).length).toBe(2);
        });
    });

    describe("render `BeerComponent` when beerlist is empty", () => {
        it("show no container data ", () => {
            const divs = beerListContainer.find("div").first();
            const para = divs.children();
            expect(para.length).toBe(1);
        });
    });


    it('renders correctly', () => {
        expect(beerListContainer).toMatchSnapshot();
    });
});
