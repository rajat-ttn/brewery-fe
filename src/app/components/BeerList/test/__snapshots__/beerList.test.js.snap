// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerList Component should render correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerList
    beerList={
        Array [
            Object {
              "beerType": "Aler Beer",
              "containerId": 1,
              "id": 1,
              "tempRange": Array [
                3,
                5,
              ],
            },
          ]
    }
    temperatureType="FAHRENHEIT"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BeerComponent
          beerContentDetail={
                    Object {
                              "beerType": "Aler Beer",
                              "containerId": 1,
                              "id": 1,
                              "tempRange": Array [
                                3,
                                5,
                              ],
                            }
          }
          temperatureType="FAHRENHEIT"
/>,
      ],
      "className": "row beerlistSection",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "beerContentDetail": Object {
            "beerType": "Aler Beer",
            "containerId": 1,
            "id": 1,
            "tempRange": Array [
              3,
              5,
            ],
          },
          "temperatureType": "FAHRENHEIT",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BeerComponent
            beerContentDetail={
                        Object {
                                    "beerType": "Aler Beer",
                                    "containerId": 1,
                                    "id": 1,
                                    "tempRange": Array [
                                      3,
                                      5,
                                    ],
                                  }
            }
            temperatureType="FAHRENHEIT"
/>,
        ],
        "className": "row beerlistSection",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "beerContentDetail": Object {
              "beerType": "Aler Beer",
              "containerId": 1,
              "id": 1,
              "tempRange": Array [
                3,
                5,
              ],
            },
            "temperatureType": "FAHRENHEIT",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
